3/test/excercise3.spec.js

```diff
@@ -0,0 +1,110 @@
+            expect(result.data.todosByState.length).toBe(3);
+        })
+    })
+    it('return undone todos', () => {
+        query({query: queryTodosByState, variables: {state: 'UNDONE'}}).then((result) => {
+            expect(result.data.todosByState.length).toBe(2);
+        })
+    })
+    it('return in progress todos', () => {
+        query({query: queryTodosByState, variables: {state: 'IN_PROGRESS'}}).then((result) => {
+            expect(result.data.todosByState.length).toBe(1);
+            expect(result.data.todosByState[0].message).toBe('Go shopping')
+        })
+    })
+})
+
+describe('Mutation: Given a new todo', () => {
+
+    const mutateCreateTodo = gql`mutation createNewTodo($id: ID, $message: String!, $state: State, $assigneeId: ID) {
+        createTodo(id: $id, message: $message, state: $state, assigneeId: $assigneeId) { id, message } 
+    }`;
+
+    describe('Add todo item with existing unique id to list', () => {
+        it('reject duplicate id entry', () => {
+            mutate({mutation: mutateCreateTodo, variables: 
+                {id: 1, message: 'Do something', state: "UNDONE", assigneeId: 1}}).then((result) => {
+                expect(result.data.createTodo).toBeNull();
+            })
+        })
+    })
+    describe('Add todo item with new unique id to list', () => {
+        it('todo item was added to list', () => {
+            mutate({mutation: mutateCreateTodo, variables: 
+                {id: 23, message: 'Do something', state: 'UNDONE', assigneeId: 1}}).then((result) => {
+                expect(result.data.createTodo).not.toBeNull();
+                expect(result.data.createTodo.id).toBe('23');
+                expect(result.data.createTodo.message).toEqual('Do something');
+            })
+        })
+    })
+})
+
+describe('Mutation: Update an existing todo', () => {
+    
+    const mutateUpdateTodo = gql`mutation updateTodo($id: ID!, $message: String, $state: State, $assigneeId: ID) {
+        updateTodo(id: $id, message: $message, state: $state, assigneeId: $ assigneeId) { id, message }}`;
+    const queryTodo = gql`query getTodo($id: ID!) { todo(id: $id) { id, message, state, assigneeId } }`;
+
+    it('todo item is updated', () => {
+        mutate({mutation: mutateUpdateTodo, variables: {id: 1, message: 'New TODO', state: 'IN_PROGRESS', assigneeId: 3}}).then((result) => {
+            expect(result.data.updateTodo).toBeTruthy();
+        })
+        query({query: queryTodo, variables: {id: 1}}).then((result) => {
+            expect(result.data.todo).toMatchObject({id: '1', message: 'New TODO', state: 'IN_PROGRESS', assigneeId: '3'})
+        })
+    })
+})
+
+describe('Mutation: Delete an existing todo', () => {
+    const mutateDeleteTodo = gql`mutation deleteTodo($id: ID!) { deleteTodo(id: $id) }`;
+    const queryTodos = gql`query getTodos {todos { id, message, state }}`;
+
+    it('todo removed from list', () => {
```

Comment:
```suggestion
    it('removes a todo', () => {
```

And implement the expectations in a similar fashion as above :point_up: